src/sentry/api/endpoints/organization_events_stats.py

```diff
@@ -266,6 +269,161 @@ def get_event_stats(
+            in the case that this request is from a dashboard widget and we're trying to split their discover dataset.
+
+            This should be removed once the discover dataset is completely split in dashboards.
+            """
+            dashboard_widget_id = request.GET.get("dashboardWidgetId", None)
+
+            def fn(
+                query_columns: Sequence[str],
+                query: str,
+                params: dict[str, str],
+                rollup: int,
+                zerofill_results: bool,
+                comparison_delta: datetime | None,
+            ) -> SnubaTSResult:
+                if not (metrics_enhanced and dashboard_widget_id):
+                    return _get_event_stats(
+                        scopedDataset,
+                        query_columns,
+                        query,
+                        params,
+                        rollup,
+                        zerofill_results,
+                        comparison_delta,
+                    )
+
+                try:
+                    widget = DashboardWidget.objects.get(id=dashboard_widget_id)
+                    does_widget_have_split = widget.discover_widget_split is not None
+
+                    if does_widget_have_split:
+                        # This is essentially cached behaviour and we skip the check
+                        if widget.discover_widget_split == DashboardWidgetTypes.ERROR_EVENTS:
+                            errors_only_query = f"({query}) AND event.type:error"
+                            return _get_event_stats(
+                                discover,
+                                query_columns,
+                                errors_only_query,
+                                params,
+                                rollup,
+                                zerofill_results,
+                                comparison_delta,
+                            )
+                        if widget.discover_widget_split == DashboardWidgetTypes.TRANSACTION_LIKE:
+                            return _get_event_stats(
+                                scopedDataset,
+                                query_columns,
+                                query,
+                                params,
+                                rollup,
+                                zerofill_results,
+                                comparison_delta,
+                            )
+                        if widget.discover_widget_split == DashboardWidgetTypes.DISCOVER:
+                            # This is a fallback for the ambiguous case.
+                            return _get_event_stats(
+                                discover,
+                                query_columns,
+                                query,
+                                params,
+                                rollup,
+                                zerofill_results,
+                                comparison_delta,
+                            )
```

Comment:
Can we reduce the code repeat for this pattern a bit, maybe set the dataset and query to a variable and then call `get_events_stats` once at the end?